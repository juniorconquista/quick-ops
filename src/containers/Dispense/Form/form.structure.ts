import * as Yup from 'yup';

export const schema = Yup.object().shape({
    type: Yup.string().required('O campo Tipo é obrigatorio'),
    title: Yup.string().required('O campo titulo da despesa é obrigatorio'),
});
