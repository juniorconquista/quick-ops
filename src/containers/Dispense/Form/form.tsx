import React, { memo } from 'react';
import { Formik, Field, Form, FieldProps } from 'formik';
import { Box } from 'quarks-ui';
import { css } from 'styled-components';

import Button from '../../../components/Button';
import Input from '../../../components/Input';
import Select from '../../../components/Select';
import DatePicker from '../../../components/DatePicker';

import Upload from '../../../components/Upload';
import Loading from '../../../components/Loading';
import useAsync from '../../../hooks/useAsync';

import { FormProps } from './form.types';
import {
    schema,
    optionsExpenseTypes,
    optionsCurrencyCode,
} from './form.structure';
import 'react-datepicker/dist/react-datepicker.css';

const Dispense: React.FC<FormProps> = (props) => {
    const addDispense = (body: any): Promise<any> => {
        const headers = new Headers();
        headers.append('Content-Type', 'application/json');
        headers.append('Accept', 'application/json');
        return fetch(`${process.env.REACT_APP_API_URL}/expense/add`, {
            method: 'POST',
            headers: headers,
            body: body,
        }).then((response) => response.json());
    };
    const { execute, status } = useAsync<any>(addDispense, false);

    return (
        <Formik
            validationSchema={schema}
            initialValues={{
                resourceUrl: '',
                expenseTypeCode: { label: 'Selecione', value: '' },
                currencyCode: { label: 'Selecione', value: '' },
                notes: '',
                cardDate: '',
                amountTotal: '',
                amountSpent: '',
            }}
            onSubmit={async (values, actions) => {
                const body = {
                    ...values,
                    expenseTypeCode: values.expenseTypeCode.value,
                    currencyCode: values.currencyCode.value,
                    cardDate: new Date(values.cardDate).getTime(),
                };
                await execute(JSON.stringify(body));
                actions.setSubmitting(false);
                actions.resetForm();
                props.handleDispense();
            }}
        >
            {({ values }) => (
                <Form style={{ width: '100%' }}>
                    <Box
                        styling="row"
                        alignItems="stretch"
                        width="100%"
                        height="100%"
                        p={{
                            mobile: '15px',
                            tablet: '15px',
                            desktop: '12px 48px',
                        }}
                        css={css({
                            '@media (max-width: 1300px)': {
                                flexDirection: 'column',
                            },
                        })}
                    >
                        <Box
                            styling="row"
                            bg="white.default"
                            width="100%"
                            flex="1"
                            css={css({
                                '@media (max-width: 1300px)': {
                                    marginBottom: '20px',
                                },
                            })}
                        >
                            <Box
                                styling="column"
                                width="100%"
                                height={{
                                    mobile: '100%',
                                    tablet: '100%',
                                    desktop: '320px',
                                }}
                                p={{
                                    mobile: '15px',
                                    tablet: '15px',
                                    desktop: '40px',
                                }}
                            >
                                <Field name="resourceUrl">
                                    {({
                                        field,
                                        form: { setFieldValue },
                                        meta,
                                    }: FieldProps) => (
                                        <Upload
                                            styling={
                                                meta.touched && meta.error
                                                    ? 'danger'
                                                    : 'base'
                                            }
                                            label="Recibo, cupom ou nota fiscal*"
                                            name={field.name}
                                            error={meta.touched && meta.error}
                                            onChange={(value) =>
                                                setFieldValue(
                                                    'resourceUrl',
                                                    value,
                                                )
                                            }
                                        />
                                    )}
                                </Field>
                            </Box>
                        </Box>
                        <Box
                            styling="column"
                            min-height="360px"
                            width="100%"
                            flex="1"
                            justifyContent="flex-start"
                            pl={{
                                mobile: '15px',
                                tablet: '15px',
                                desktop: '48px',
                            }}
                            css={css({
                                '@media (max-width: 1300px)': {
                                    paddingLeft: '0px',
                                },
                            })}
                        >
                            <Box
                                styling="grid"
                                gridTemplateColumns="repeat(2, 1fr)"
                                gridGap={20}
                                width="100%"
                            >
                                <Field name="expenseTypeCode">
                                    {({
                                        field,
                                        form: { setFieldValue },
                                        meta,
                                    }: FieldProps) => {
                                        const error =
                                            meta.touched &&
                                            meta.error &&
                                            //@ts-ignore
                                            meta.error.value;
                                        return (
                                            <Select
                                                styling={
                                                    error ? 'danger' : 'base'
                                                }
                                                placeholder="Selecione"
                                                label="Tipo*"
                                                options={optionsExpenseTypes}
                                                name={field.name}
                                                onChange={(value) =>
                                                    setFieldValue(
                                                        'expenseTypeCode',
                                                        value,
                                                    )
                                                }
                                                onBlur={field.onBlur}
                                                error={error}
                                                value={field.value}
                                            />
                                        );
                                    }}
                                </Field>
                                <Field name="currencyCode">
                                    {({
                                        field,
                                        form: { setFieldValue },
                                        meta,
                                    }: FieldProps) => {
                                        const error =
                                            meta.touched &&
                                            meta.error &&
                                            //@ts-ignore
                                            meta.error.value;
                                        return (
                                            <Select
                                                styling={
                                                    error ? 'danger' : 'base'
                                                }
                                                placeholder="Selecione"
                                                label="Moeda*"
                                                options={optionsCurrencyCode}
                                                name={field.name}
                                                onChange={(value) =>
                                                    setFieldValue(
                                                        'currencyCode',
                                                        value,
                                                    )
                                                }
                                                onBlur={field.onBlur}
                                                error={error}
                                                value={field.value}
                                            />
                                        );
                                    }}
                                </Field>
                            </Box>
                            <Field name="notes">
                                {({ field, meta }: FieldProps) => (
                                    <Input
                                        styling={
                                            meta.touched && meta.error
                                                ? 'danger'
                                                : 'base'
                                        }
                                        placeholder="Digite..."
                                        label="Título da despesa*"
                                        name={field.name}
                                        value={field.value}
                                        onChange={field.onChange}
                                        onBlur={field.onBlur}
                                        error={meta.touched && meta.error}
                                    />
                                )}
                            </Field>
                            <Field name="cardDate">
                                {({
                                    field,
                                    form: { setFieldValue },
                                    meta,
                                }: FieldProps) => (
                                    <DatePicker
                                        styling={
                                            meta.touched && meta.error
                                                ? 'danger'
                                                : 'base'
                                        }
                                        label="Data do comprovante*"
                                        placeholderText="DD/MM/YYYY"
                                        selected={field.value}
                                        name={field.name}
                                        onBlur={field.onBlur}
                                        onChange={(value) =>
                                            setFieldValue('cardDate', value)
                                        }
                                        error={meta.touched && meta.error}
                                    />
                                )}
                            </Field>
                            {values.currencyCode.value !== '' && (
                                <Box
                                    styling="grid"
                                    gridTemplateColumns="repeat(2, 1fr)"
                                    gridGap={20}
                                    width="100%"
                                >
                                    <Field name="amountTotal">
                                        {({ field, meta }: FieldProps) => (
                                            <Input
                                                styling={
                                                    meta.touched && meta.error
                                                        ? 'danger'
                                                        : 'base'
                                                }
                                                type="number"
                                                placeholder="Digite..."
                                                label="Valor da nota / cupom*"
                                                name={field.name}
                                                value={field.value}
                                                onChange={field.onChange}
                                                onBlur={field.onBlur}
                                                error={
                                                    meta.touched && meta.error
                                                }
                                            />
                                        )}
                                    </Field>
                                    <Field name="amountSpent">
                                        {({ field, meta }: FieldProps) => (
                                            <Input
                                                styling={
                                                    meta.touched && meta.error
                                                        ? 'danger'
                                                        : 'base'
                                                }
                                                type="number"
                                                placeholder="Digite..."
                                                label="Valor a ser considerado*"
                                                name={field.name}
                                                value={field.value}
                                                onChange={field.onChange}
                                                onBlur={field.onBlur}
                                                error={
                                                    meta.touched && meta.error
                                                }
                                            />
                                        )}
                                    </Field>
                                </Box>
                            )}
                        </Box>
                    </Box>
                    <Box
                        styling="row"
                        height="77px"
                        mt="20px"
                        bg="white.default"
                        width="100%"
                        justifyContent="flex-end"
                    >
                        <Button
                            type="button"
                            styling="base"
                            text="Cancelar"
                            onClick={props.handleDispense}
                        />
                        <Button
                            type="submit"
                            styling="success"
                            text="Salvar"
                            m="0px 24px"
                        />
                    </Box>
                    {status === 'pending' && <Loading />}
                </Form>
            )}
        </Formik>
    );
};

export default memo(Dispense);
