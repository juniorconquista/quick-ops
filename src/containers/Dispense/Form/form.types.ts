export interface FormProps {
    handleDispense: () => void;
}
