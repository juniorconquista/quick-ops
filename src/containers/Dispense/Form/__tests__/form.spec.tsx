import React from 'react';
import { cleanup, render } from '@testing-library/react';

import ThemeProvider from '../../../../utils/ThemeProviderTest';
import Form from '../form';
import 'jest-styled-components';

const props = {
    handleDispense: jest.fn(),
};

describe('Dispense Form container', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('tests basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Form {...props} />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });
});
