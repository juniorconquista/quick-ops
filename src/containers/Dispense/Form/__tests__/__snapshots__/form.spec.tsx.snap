// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispense Form container tests basic snapshot 1`] = `
<DocumentFragment>
  .c4 {
  color: #FFFFFF;
  font-weight: bold;
  margin: 12px 0px;
}

.c6 {
  color: #f8f9fa;
  font-size: 1rem;
  color: #848a91;
  text-align: left;
  margin-top: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  height: 100%;
}

.c0:focus {
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
}

.c1:focus {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
}

.c3:focus {
  outline: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c7:focus {
  outline: none;
}

.c8 {
  display: grid;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-template-columns: repeat(2,1fr);
  grid-gap: 20px;
  width: 100%;
}

.c8:focus {
  outline: none;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  margin-top: 20px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 77px;
  width: 100%;
}

.c12:focus {
  outline: none;
}

.c13 {
  height: 36px;
  border-radius: 6px;
  color: #6b7480 !important;
  border: 1px solid;
  border-color: #6b7480;
  padding: 0px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  cursor: pointer;
  -webkit-transition: boxShadow .25s ease;
  transition: boxShadow .25s ease;
  font-family: Nunito Sans,sans-serif;
  color: #FFFFFF;
}

.c13 svg {
  height: 15px;
  width: 15px;
  margin-right: 5px;
  fill: #6b7480;
}

.c13:hover,
.c13:focus {
  box-shadow: 0 0 2px 3px #6b748090;
}

.c14 {
  height: 36px;
  border-radius: 6px;
  color: #FFFFFF;
  border: 1px solid;
  border-color: #51c1c3;
  padding: 0px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #51c1c3;
  cursor: pointer;
  -webkit-transition: boxShadow .25s ease;
  transition: boxShadow .25s ease;
  font-family: Nunito Sans,sans-serif;
  color: #FFFFFF;
  margin: 0px 24px;
}

.c14 svg {
  height: 15px;
  width: 15px;
  margin-right: 5px;
  fill: #FFFFFF;
}

.c14:hover,
.c14:focus {
  box-shadow: 0 0 2px 3px #51c1c390;
}

.c10 {
  width: 100%;
}

.c10 label {
  margin: 12px 0px;
  color: #FFFFFF;
  font-weight: bold;
  display: block;
}

.c10 input {
  height: 36px;
  border-radius: 6px;
  color: #6b7480 !important;
  border: 1px solid;
  border-color: #dfe2e6;
  padding: 12px 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  cursor: pointer;
  width: 100%;
  -webkit-transition: boxShadow .25s ease;
  transition: boxShadow .25s ease;
  font-family: Nunito Sans,sans-serif;
}

.c10 input:hover,
.c10 input:focus {
  box-shadow: 0 0 2px 3px #6b748090;
  outline: none;
}

.c9 {
  width: 100%;
}

.c9 label {
  margin: 12px 0px;
  color: #FFFFFF;
  font-weight: bold;
  display: block;
}

.c9 .select__input {
  color: #6b7480;
  font-family: Nunito Sans,sans-serif;
  font-weight: normal;
}

.c9 .select__input > div {
  font-family: Nunito Sans,sans-serif !important;
  font-weight: normal !important;
}

.c9 .select__control {
  border: 1px solid;
  border-radius: 6px;
  background-color: #25262a;
  border-color: #dfe2e6;
  font-family: Nunito Sans,sans-serif;
  height: 36px;
  -webkit-transition: inherit;
  transition: inherit;
}

.c9 .select__control:hover {
  box-shadow: 0 0 2px 3px #6b748090;
  outline: none;
  border: 1px solid;
  border-color: #dfe2e6;
}

.c9 .select__placeholder {
  color: #6b7480;
  font-family: Nunito Sans,sans-serif;
}

.c9 .select__single-value {
  color: #6b7480;
  font-family: Nunito Sans,sans-serif;
}

.c9 .select__control--is-focused {
  box-shadow: 0 0 2px 3px #6b748090;
  border-color: #dfe2e6;
}

.c9 .select__menu {
  background-color: #25262a;
  border: 1px solid;
  border-color: #dfe2e6;
  margin-top: 4px;
  overflow: hidden;
}

.c9 .select__menu-list {
  color: #6b7480;
  font-family: Nunito Sans,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c9 .select__option {
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  line-height: normal;
}

.c9 .select__option:hover {
  background-color: #313438;
}

.c9 .select__option--is-focused {
  background-color: #313438;
}

.c9 .select__option--is-selected {
  background-color: #51c1c3;
}

.c11 {
  width: 100%;
}

.c11 .react-datepicker-wrapper {
  width: 100%;
}

.c11 .react-datepicker-wrapper input {
  height: 36px;
  border-radius: 6px;
  color: #6b7480 !important;
  border: 1px solid;
  border-color: #dfe2e6;
  padding: 12px 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  cursor: pointer;
  width: 100%;
  -webkit-transition: boxShadow .25s ease;
  transition: boxShadow .25s ease;
  font-family: Nunito Sans,sans-serif;
}

.c11 .react-datepicker-wrapper input:hover,
.c11 .react-datepicker-wrapper input:focus {
  box-shadow: 0 0 2px 3px #6b748090;
  outline: none;
}

.c11 .react-datepicker__current-month,
.c11 .react-datepicker__day-name,
.c11 .react-datepicker__day {
  font-family: Nunito Sans,sans-serif;
  color: #f8f9fa;
}

.c11 .react-datepicker__triangle {
  display: none;
}

.c11 .react-datepicker__header {
  background-color: #25262a;
}

.c11 .react-datepicker__month {
  background-color: #313438;
  margin: 0px;
  padding: 0.4rem;
  overflow: hidden;
}

.c11 .react-datepicker__day:hover {
  background-color: #25262a;
  color: #f8f9fa;
}

.c11 .react-datepicker__day--selected {
  background-color: #51c1c3;
  color: #FFFF;
}

.c11 .react-datepicker__day--selected:hover {
  background-color: #51c1c3;
  color: #FFFF;
}

.c5 {
  height: 208px;
  width: 100%;
  border-radius: 6px;
  color: #6b7480;
  border: 1px dashed;
  border-color: #dfe2e6;
  padding: 0px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-color: #25262a;
  cursor: pointer;
  -webkit-transition: boxShadow .25s ease;
  transition: boxShadow .25s ease;
}

.c5 p {
  border: 1px solid;
  border-color: #dfe2e6;
  border-radius: 6px;
  color: #6b7480;
  height: 46px;
  padding: 0px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Nunito Sans,sans-serif;
  text-align: center;
  margin-bottom: 44px;
}

.c5:hover,
.c5:focus {
  outline: none;
}

.c5:hover p,
.c5:focus p {
  box-shadow: 0 0 2px 3px #6b748090;
}

@media screen and (min-width:0rem) {
  .c0 {
    padding: 15px;
  }
}

@media screen and (min-width:37.5rem) {
  .c0 {
    padding: 15px;
  }
}

@media screen and (min-width:62rem) {
  .c0 {
    padding: 12px 48px;
  }
}

@media (max-width:1300px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:1300px) {
  .c1 {
    margin-bottom: 20px;
  }
}

@media screen and (min-width:0rem) {
  .c2 {
    padding: 15px;
  }
}

@media screen and (min-width:37.5rem) {
  .c2 {
    padding: 15px;
  }
}

@media screen and (min-width:62rem) {
  .c2 {
    padding: 40px;
  }
}

@media screen and (min-width:0rem) {
  .c2 {
    height: 100%;
  }
}

@media screen and (min-width:37.5rem) {
  .c2 {
    height: 100%;
  }
}

@media screen and (min-width:62rem) {
  .c2 {
    height: 320px;
  }
}

@media screen and (min-width:0rem) {
  .c7 {
    padding-left: 15px;
  }
}

@media screen and (min-width:37.5rem) {
  .c7 {
    padding-left: 15px;
  }
}

@media screen and (min-width:62rem) {
  .c7 {
    padding-left: 48px;
  }
}

@media (max-width:1300px) {
  .c7 {
    padding-left: 0px;
  }
}

<form
    action="#"
    style="width: 100%;"
  >
    <div
      class="c0"
      height="100%"
      width="100%"
    >
      <div
        class="c1"
        width="100%"
      >
        <div
          class="c2"
          height="[object Object]"
          width="100%"
        >
          <div
            class="c3"
            width="100%"
          >
            <label
              class="c4"
              color="text.contrast"
              font-weight="bold"
            >
              Recibo, cupom ou nota fiscal*
            </label>
            <div
              class="c5"
              tabindex="0"
            >
              <input
                accept="image/jpeg, image/png"
                autocomplete="off"
                name="resourceUrl"
                style="display: none;"
                tabindex="-1"
                type="file"
              />
              <p>
                Selecione um arquivo do seu computador
              </p>
            </div>
            <p
              class="c6"
              color="#848a91"
            >
              A imagem deve estar no formato JPG ou PNG.
            </p>
          </div>
        </div>
      </div>
      <div
        class="c7"
        width="100%"
      >
        <div
          class="c8"
          width="100%"
        >
          <div
            class="c3"
            width="100%"
          >
            <label
              class="c4"
              color="text.contrast"
              font-weight="bold"
            >
              Tipo*
            </label>
            <div
              class="c9 css-2b097c-container"
              id="expenseTypeCode"
            >
              <div
                class="select__control css-yk16xz-control"
              >
                <div
                  class="select__value-container select__value-container--has-value css-g1d714-ValueContainer"
                >
                  <div
                    class="select__single-value css-1uccc91-singleValue"
                    style="opacity: 1; transition: opacity 1ms;"
                  >
                    Selecione
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class="select__input"
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-2-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class="select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c3"
            width="100%"
          >
            <label
              class="c4"
              color="text.contrast"
              font-weight="bold"
            >
              Moeda*
            </label>
            <div
              class="c9 css-2b097c-container"
              id="currencyCode"
            >
              <div
                class="select__control css-yk16xz-control"
              >
                <div
                  class="select__value-container select__value-container--has-value css-g1d714-ValueContainer"
                >
                  <div
                    class="select__single-value css-1uccc91-singleValue"
                    style="opacity: 1; transition: opacity 1ms;"
                  >
                    Selecione
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class="select__input"
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-3-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class="select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c10"
        >
          <label
            for="notes"
          >
            Título da despesa*
          </label>
          <input
            id="notes"
            name="notes"
            placeholder="Digite..."
            type="text"
            value=""
          />
        </div>
        <div
          class="c3"
          width="100%"
        >
          <label
            class="c4"
            color="text.contrast"
            font-weight="bold"
          >
            Data do comprovante*
          </label>
          <div
            class="c11"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  id="cardDate"
                  placeholder="DD/MM/YYYY"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
      height="77px"
      width="100%"
    >
      <button
        class="c13"
        color="#FFFFFF"
        type="button"
      >
        Cancelar
      </button>
      <button
        class="c14"
        color="#FFFFFF"
        type="submit"
      >
        Salvar
      </button>
    </div>
  </form>
</DocumentFragment>
`;
